common/models/attack-model.ts

```diff
@@ -172,7 +172,17 @@ export class AttackModel {
 		return this
 	}
 
-	/** Reduces the damage the attack does */
+	/** Reduces the base damage of the attack (before multiplication) */
+	public removeDamage(source: AttackerEntity, amount: number) {
```

Comment:
I think this function should actually be called `subtractDamage` so it's clear it's the opposite of add. I think that would clear up the confusion I had a bit.

Reply:
Or maybe keeping this as `removeDamage` is fine, ill just go back and rename the other function since its kinda on the edge of what fits in this PR.